yt_dlp/extractor/tennistv.py

```diff
@@ -31,84 +33,126 @@ class TennisTVIE(InfoExtractor):
     }
     _NETRC_MACHINE = 'tennistv'
 
-    def _login(self):
-        username, password = self._get_login_info()
-        if not username or not password:
-            raise ExtractorError('No login info available, needed for using %s.' % self.IE_NAME, expected=True)
+    ACCESS_TOKEN = None
+    REFRESH_TOKEN = None
+    PARTNER_ID = 3001482
+    _FORMAT_URL = 'https://open.http.mp.streamamg.com/p/{partner}/sp/{partner}00/playManifest/entryId/{entry}/format/applehttp/protocol/https/a.m3u8?ks={session}'
+    headers = {
+        'origin': 'https://www.tennistv.com',
+        'referer': 'https://www.tennistv.com/',
+        'content-Type': 'application/x-www-form-urlencoded'
+    }
+
+    def _perform_login(self, username, password):
 
+        BASE_URL = 'https://sso.tennistv.com/auth/realms/TennisTV/protocol/openid-connect/auth'
```

Comment:
makes more sense to put `https://sso.tennistv.com/auth/realms/TennisTV/protocol` in a class var instead imo. This is also used in `get_token`